pkg/metallb/frrnodestate.go

```diff
@@ -0,0 +1,112 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// FrrNodeStateBuilder provides struct for FrrNodeState object which contains connection to cluster and
+// frrconfiguration definitions.
+type FrrNodeStateBuilder struct {
+	// Dynamically discovered FrrNodeState object.
+	Objects *frrtypes.FRRNodeState
+	// apiClient opens api connection to the cluster.
+	apiClient runtimeClient.Client
+	// nodeName defines on what node FrrNodeState resource should be queried.
+	nodeName string
+	// nsName defines metallb operator namespace.
+	nsName string
+	// errorMsg used in discovery function before sending api request to cluster.
+	errorMsg string
+}
+
+// NewFrrNodeStateBuilder creates new instance of FrrNodeStateBuilder.
+func NewFrrNodeStateBuilder(apiClient *clients.Settings, nodeName, nsname string) *FrrNodeStateBuilder {
+	glog.V(100).Infof(
+		"Initializing new FrrNodeStateBuilder structure with the following params: %s, %s",
+		nodeName, nsname)
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient cannot be nil")
+
+		return nil
+	}
+
+	err := apiClient.AttachScheme(frrtypes.AddToScheme)
+	if err != nil {
+		glog.V(100).Infof("Failed to add metallb scheme to client schemes")
+
+		return nil
+	}
+
+	builder := &FrrNodeStateBuilder{
+		apiClient: apiClient.Client,
+		nodeName:  nodeName,
+		nsName:    nsname,
+	}
+
+	if nodeName == "" {
+		glog.V(100).Infof("The name of the nodeName is empty")
+
+		builder.errorMsg = "FrrNodeState 'nodeName' is empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the FrrNodeState is empty")
+
+		builder.errorMsg = "FrrNodeState 'nsname' is empty"
+	}
+
+	return builder
+}
+
+// Discover method gets the FrrNodeState items and stores them in the FrrNodeStateBuilder struct.
+func (builder *FrrNodeStateBuilder) Discover() error {
```

Comment:
Why not just use a Pull function instead?

Reply:
Will also remove the need for a NewBuilder function

Reply:
Hi Trey I added a pull/exist/get/ and removed the NewBuilder.